### ISSUE TITLE
Focusing search textbox after clearing it

### ISSUE THREAD

**Issue by h-h-h-h** [davidhealey/waistline#493]
When you're in the food list, search, select and then want to search again, clearing the textbox by pressing `←` to the left of the textbox, you have to additionally tap the textbox to focus it. It would be great if the textbox was automatically focused after clearing it. EDIT: At least when the keyboard was visible when tapping `←`. If it was not, one possibly might have wanted to just show the unfiltered list, without reducing space by the keyboard.

Also, I associate `←` with navigation. `×` would be better suited as an icon in my opinion.

**Comment by EmilJunker** [comment]
> It would be great if the textbox was automatically focused after clearing it.

This is not so easy because the `←` button and its behavior is part of [framework7](https://framework7.io/docs/searchbar), and I don't know if it can be changed.

> EDIT: At least when the keyboard was visible when tapping `←`. If it was not, one possibly might have wanted to just show the unfiltered list, without reducing space by the keyboard.

You rightly point out that automatically showing the keyboard when the `←` button is tapped would have unwanted side effects. Hence, it should only happend when the search box is already focused. This would then allow the following workflow:

- Step 1: enter a query into the search box
- Step 2: select an item from the results list (this dismisses the keyboard)
- Step 3: tap the search box again to show the keyboard
- Step 4: tap the `←` button to clear the search box (and suppose this doesn't dismiss the keyboard)

But how is this any easier than doing the following (same steps but in a slightly different order):

- Step 1: enter a query into the search box
- Step 2: select an item from the results list (this dismisses the keyboard)
- Step 3: tap the `←` button to clear the search box
- Step 4: tap the search box again to show the keyboard

In other words, the change you are proposing seems a bit unnecessary.

> Also, I associate `←` with navigation. `×` would be better suited as an icon in my opinion.

Again, this is the default button coming from framework7. I'm not sure if/how it can be changed.

**Comment by h-h-h-h** [comment]
> > EDIT: At least when the keyboard was visible when tapping ←. If it was not, one possibly might have wanted to just show the unfiltered list, without reducing space by the keyboard.
>
> You rightly point out that automatically showing the keyboard when the ← button is tapped would have unwanted side effects.

When I added this, I forgot that the keyboard already was dismissed when you're about to tap `←`.

> This would then allow the following workflow:

What I actually wanted to get rid of is having to tap twice, which is much more intuitive and less tedious for that common use case of mine.

The question is: If the foods are filtered by search box text and the keyboard currently isn't displayed, what would the user most likely want to do next after pressing `←`? It might be different for different people. I can only advocate for doing the next search box filtering, which is a very common use case of mine. I can't judge on how common it would be for users to want to go through the unfiltered list next. Maybe, going through the unfiltered list could be expected to have been done at the start before any more specific filtering? If you don't think so, we'd need a setting about whether to focus the search box (best default in my opinion: do focus it).

I found the following other case for a textbox with clear button: In VLC for Android, if you enter text to search under the "Audio" tab, hide the keyboard so that the search box is defocused, and press the `x` button in the search box, the search box is focused. Although, granted, there's also a `←` button which "navigates" back to the unfiltered view of the list.

**Comment by EmilJunker** [comment]
I see, now I understand what you mean.

> As I can see in your Framework7 link, there already seems to be a provision for that: `input-clear-button`

Yes, the built-in clear button seems to do exactly what we want. So this is what we'll go with.
